client/src/containers/methodology/index.tsx

```diff
@@ -0,0 +1,77 @@
+"use client";
+
+import { useMemo, useRef } from "react";
+
+import { useSetAtom } from "jotai";
+
+import { useScrollSpy } from "@/hooks/use-scroll-spy";
+
+import METHODOLOGY_SECTIONS from "@/containers/methodology/sections";
+import MethodologySidebar from "@/containers/methodology/sidebar";
+import { methodologyStepAtom } from "@/containers/methodology/store";
+import { getSidebarNavItemAriaId } from "@/containers/sidebar/sidebar-navigation";
+
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { ScrollArea } from "@/components/ui/scroll-area";
+import { SidebarTrigger } from "@/components/ui/sidebar";
+
+export default function Methodology() {
+  const ref = useRef<HTMLDivElement>(null);
+  const setMethodologyStep = useSetAtom(methodologyStepAtom);
+  useScrollSpy({
+    id: "methodology-sections-container",
+    containerRef: ref,
+    setCurrentStep: setMethodologyStep,
+    options: {
+      threshold: 0.1,
+      rootMargin: "0% 0% -70% 0%",
+    },
+  });
+  const navItems = useMemo(() => {
+    return METHODOLOGY_SECTIONS.map(({ id, title, href }) => ({
+      id,
+      label: title,
+      href,
+    }));
+  }, []);
```

Comment:
Isn't `METHODOLOGY_SECTIONS` static content? I would move this parse out of the Methodology function as there is nothing the parse depends on.